frontend/js/src/user/Dashboard.tsx

```diff
@@ -468,6 +471,12 @@ export default function Listen() {
               />{" "}
               MusicBrainz
             </Link>
+            {user && !isCurrentUsersPage && (
+              <ReportUserButton
+                user={user}
+                alreadyReported={already_reported_user}
+              />
+            )}
```

Comment:
We should check if the user is logged in or not, then only they should be allowed to report a user. 
The best way is to check if the user is logged in the modal, and accordingly redirect user to the login page first if they're not logged in.

Reply:
I decided to show a link to the user in a toast message, in case they were doing something else on the page they didn't want to stop, like playing music.
Also made the link redirect back to the current page, which is something we should probably do in other cases in the website where we require login for actions, but that's for another day.